### PR TITLE
migrations were not, in fact, idempotent, because using .First reorde…

### LINKED ISSUE #78: Error migrating database: Error 1050: Table 'cloud_operations' already exists
I encountered this error after clicking "apply changes" for v2.1.1 of the GCP tile / service broker.  Here are the steps which would trigger this:

(1) Freshly installed PCF on GCP (Ops Man v. 1.9.0.0, ERT v. 1.8.20-build.7)
(2) Configured GCP tile, clicked "apply changes"
(3) Hit an error with quota (Error 403: Quota 'CPUS' exceeded.)
(4) Requested and was granted quota increase
(5) Clicked "apply changes" again
(6) Succeeded in deploying GCP tile
(7) Added a new tile, MySQL 1.8.0
(8) Clicked "apply changes"
(9) Exited with an error (see below)

```2016-12-11T11:42:33.29+0000 [APP/0]      OUT [35m(Error 1050: Table 'cloud_operations' already exists)[0m 
2016-12-11T11:42:33.29+0000 [APP/0]      OUT [33m[2016-12-11 11:42:33][0m [31;1m [0m
2016-12-11T11:42:33.29+0000 [APP/0]      ERR panic: Error migrating database: Error 1050: Table 'cloud_operations' already exists
2016-12-11T11:42:33.29+0000 [APP/0]      ERR goroutine 1 [running]:
2016-12-11T11:42:33.29+0000 [APP/0]      ERR panic(0x558e7a3aae00, 0xc82000bc60)
2016-12-11T11:42:33.29+0000 [APP/0]      ERR /tmp/cache/go1.6.3/go/src/runtime/panic.go:481 +0x3ea
2016-12-11T11:42:33.29+0000 [APP/0]      ERR gcp-service-broker/db_service.New.func1()
2016-12-11T11:42:33.29+0000 [APP/0]      ERR /tmp/app/.heroku/go/src/gcp-service-broker/db_service/db_service.go:37 +0x1d4
2016-12-11T11:42:33.29+0000 [APP/0]      ERR sync.(*Once).Do(0x558e7a985980, 0xc820557d70)
2016-12-11T11:42:33.29+0000 [APP/0]      ERR /tmp/cache/go1.6.3/go/src/sync/once.go:44 +0xe6
2016-12-11T11:42:33.29+0000 [APP/0]      ERR gcp-service-broker/db_service.New(0x7ff015669108, 0xc8201fe180, 0xc8202468c0)
2016-12-11T11:42:33.29+0000 [APP/0]      ERR /tmp/app/.heroku/go/src/gcp-service-broker/db_service/db_service.go:39 +0x4e
2016-12-11T11:42:33.29+0000 [APP/0]      ERR main.main()
2016-12-11T11:42:33.29+0000 [APP/0]      ERR /tmp/app/.heroku/go/src/gcp-service-broker/server.go:37 +0x31e
2016-12-11T11:42:33.30+0000 [APP/0]      OUT Exit status 2
2016-12-11T11:42:33.31+0000 [CELL/0]     OUT Exit status 0```

### PR DESCRIPTION
**Pull request by ccemeraldeyes**
…rs the migration rows. This fixes #78 and adds unit tests for migrations

### CODE REVIEW

**Comment by ccemeraldeyes** [comment]
@mgoddard please review

File: db_service/db_service_test.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+			err = RunMigrations(DbConnection)
+			Expect(err).NotTo(HaveOccurred())
+			err = RunMigrations(DbConnection)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+	})
+
+	AfterEach(func() {
+		dropTestDatabase()
```
Comment (johnsonj) [inline db_service/db_service_test.go]: will this run even if a test fails/panics? If not you may want to add a check for it when creating the db

File: db_service/migrations.go [hunk 2]
```diff
@@ -212,11 +212,11 @@ func RunMigrations(db *gorm.DB) error {
 
 	// if we've run any migrations before, we should have a migrations table, so find the last one we ran
 	if db.HasTable("migrations") {
-		lastMigration := models.Migration{}
-		if err := db.Order("created_at desc").First(&lastMigration).Error; err != nil {
+		var storedMigrations []models.Migration
+		if err := db.Order("id desc").Find(&storedMigrations).Error; err != nil {
```
Comment (johnsonj) [inline db_service/migrations.go]: does this ID correspond to the key of [this map](https://github.com/GoogleCloudPlatform/gcp-service-broker/blob/master/db_service/migrations.go#L36)? Or is it incremented by mysql? If it's the former I recommend either controlling it or adding a new field that is controlled by you for sorting.

Comment (johnsonj) [inline db_service/migrations.go]: Looks like it is [auto incremented](https://github.com/GoogleCloudPlatform/gcp-service-broker/blob/master/db_service/migrations.go#L93). Recommend changing this to sort on `migraqtion_id`

**Pull request merged**